### DISCUSSION TITLE
Can you share the stack with other developers?

### DISCUSSION THREAD

**Discussion by amuttsch** [git-town/git-town#4149] [General]
We are developing features in pair programming. We use git town to maintain mutliple branches. 

Can we somehow share the parent / child relations of branches with others or it this strictly local? For example if I check out a branch `some-feature` from a colleague, I have to ask or check the pull request if its target is `main` or `feature-1`. So when we run `git sync` it fetches and rebases the correct branches? 

Esentially what I see with `git town branch` I want to share with other developers on my team.

Thanks!

**Comment by kevgo** [comment]
With the release of [Git Town 16.3](https://github.com/git-town/git-town/releases/tag/v16.3.0) we introduced a new feature that improves how `git town sync` handles branches without a known parent. Now, if `git town sync` encounters a branch where the parent branch is undefined locally, it automatically checks for an open pull request and sets the parent branch based on that PR.

This means that if you (or your team) create (draft) pull requests for all branches in your stack, others can easily check them out using `git town switch -a` and sync everything seamlessly—no need to manually input the parent branch.

I just realized this isn't yet documented on the [Git Town website](https://www.git-town.com), but I'll be adding it shortly. Thanks for bringing this up!

**Reply by amuttsch** [reply]
That works perfectly for us, thank you!

**Reply by kevgo** [reply]
Great to hear! I have created https://github.com/git-town/git-town/issues/4157 to add this information to the website so that it is accessible to everyone.

**Comment by davidolrik** [comment]
@kevgo  Couldn't this "without a known parent" be solved using `git reflog` looking for the nearest parent?
and thereby eliminating the need for manual "parent book-keeping"

**Reply by kevgo** [reply]
To my knowledge, the reflog tracks the history of references and branch movements only for your local repo clone. It isn't shared with the remote (origin) or with other developers' clones. This means each developer's reflog is isolated to their environment, which makes it tricky to use for shared state like parent branch tracking.

Do you know of any workaround to make the reflog shareable across repos? If there is a way, it could be an interesting alternative to our current explicit tracking.

**Reply by davidolrik** [reply]
Sorry my bad, I ment to write `git ref-list`, as `git reflog` is indeed local as it details my actions in my repo - definitely not for sharing.

But I did read a [blog post](https://dev.to/foster/a-better-way-to-visualize-stacked-branches-in-git-1pl8) by [Greg Foster](https://dev.to/foster), one of the devs of Graphite, where he details how they compute branch relationships using `git ref-list` - it even includes a link to their implementation. (The link is in the post broken, here is a working [one](https://github.com/searleser97/graphite-cli/blob/f386c2fc8d2619ec5ef942eac31557852f70a762/src/lib/git-refs/branch_relations.ts#L86-L94))

**Reply by kevgo** [reply]
Wow, that's a really clever approach&mdash;thanks for sharing! 🙌

This could definitely help in identifying unknown parent branches, assuming the parent branch is already pushed to origin (which seems like a fair assumption). Implementing this would be a huge improvement for the user experience and would also reduce the likelihood of errors, like accidentally selecting the wrong parent branch. 🚀

**Reply by kevgo** [reply]
I have created https://github.com/git-town/git-town/issues/4230 to track implementation of this idea.
